### ISSUE TITLE
bug in python swig wrapper generation

### ISSUE THREAD

**Issue by eendebakpt** [swig/swig#404]
When I have a function with a negative default argument, the default value is not mapped to the corresponding Python function. Below is a minimal example. This generates the following python function:

```
def swigtest(alpha, beta=0, funny):
    return _test.swigtest(alpha, beta, funny)
```
This function is not valid Python, since there are arguments with default values before the argument `funny`.

```
### test.i ###
%{
#define SWIG_FILE_WITH_INIT
%}
%module test
%{
#include "test.h"
%}
%include "test.h"

### test.h ###
#include <stdio.h>
void swigtest(int alpha, double beta=0, int funny = -1);

### test.cpp ###
#include <stdio.h>

void swigtest(int alpha, double beta=0, int funny = -1)
{
printf("hello\n");
return;
}
```

**Comment by vadz** [comment]
I think should have been fixed by adbe3f2e779a000d2688e758d3530752b24000bf, i.e. already fixed in 3.0.4. At the very least I can't reproduce it with the latest master:
```
%module da

%inline %{
double swigtest(int alpha, double beta=0, int funny = -1) { return alpha*beta + funny; }
%}
```
works as expected.

**Comment by eendebakpt** [comment]
I can confirm that with swig 3.0.5 the problem does not occur any more (the bug was present in 3.0.2).
